### DISCUSSION TITLE
CCU Programm startet nun - Aber 'Wenn' Bedingung muss Leeres Feld enthalten

### DISCUSSION THREAD

**Discussion by meistermolli** [thkl/hap-homematic#488] [Ideas]
Issue Update:

**ACHTUNG: Bitte den Hinweis von @peyto79 beachten, dass ein Zentralen Progamm ohne wenn Funktion nach einem CCU Neustart starten könnte!!!**

habe einen Einfluss auf die Bedienbarkeit des Homematic Programs in Homekit gefunden.
- Vielleicht ist das aber für andere von uns selbstverständlich. Für mich war es das leider bisher nicht.

Festlegung der "Wenn" Bedingung im CCU Programm muss entsprechend ausgeführt sein:

**Das Program habe ich in mehreren Versuchen so angelegt:**
```
1. Wenn
  ein Taster gedrückt wird
   oder
      ein Virtueller Taster gedrückt wird (in Homekit angelegt, da die Programme von mir nicht starten konnten)
..............
weiterer versuch:
2. Wenn
  ein Taster gedrückt wird
   oder
      ein Virtueller Taster (in Homekit angelegt, da die Programme von mir nicht starten konnten)
   oder
      leer
..............
```
![AF510E2F-2CCB-462A-8128-733D487CD0A5](https://user-images.githubusercontent.com/55836556/103468265-71b34680-4d57-11eb-8a6b-ec0e28e86132.jpeg)

Die beiden Schalten in Homekit nicht, wenn ich es als Program über HAP Homematic einbinde.
Siehe weiter oben im Issue,
Im Programm enthaltene Taster /  virtuelle Taster und die Bedienung über den Programm Button in der Webui funktionieren.
![36E94503-BE86-4E10-9C34-3B9D5BAF1699](https://user-images.githubusercontent.com/55836556/103468379-c905e680-4d58-11eb-80b9-eac0263880f8.jpeg)

Auch Skript testen Funktioniert:
Skript Testen: Funktioniert - TRUE 
object o = dom.GetObject(ID_PROGRAMS).Get('Programm Xhier Programmnamen eintragenX');if (o){Write(o.ProgramExecute());} else {Write('program not found by rega');}
____________________________________________________________________________________

LÖSUNG
____________________________________________________________________________________

`
3. Wenn Bedingung muss mindestens 1x ein leeres Leeres Feld in der 1. Ebene der Bedingung Aufweisen `
![A449D648-3701-4C55-892D-30FB3939E33E](https://user-images.githubusercontent.com/55836556/103468271-8099f900-4d57-11eb-945c-bd0f3b4b33e7.jpeg)

![DAEF2DBC-FD37-4178-B2B8-EA6EE74FB342](https://user-images.githubusercontent.com/55836556/103468566-a1b01900-4d5a-11eb-9c46-f5f62f2b3055.jpeg)

**Es muss mindestens eine Leere Wenn Bedingung eingepflegt sein (in der Ersten Ebene der "Wenn" Bedingung), damit das Programm auch über Homekit gestartet werden kann**

Ist das generell so und bei mir ein Verständnisproblem?
Dann kopiere ich mir einfach die Entsprechenden Programme und füge die leere Wenn Bedingung überall dazu.
    

    **--- Kann jemand im Wiki die entsprechende Info hinzufügen, falls das nur mit einer leeren Wenn Bedingung funktioniert?**

Gruß meistermolli

_Originally posted by @meistermolli in https://github.com/thkl/hap-homematic/issues/229#issuecomment-753540990_

**Comment by Peyto79** [comment]
Hallo,

die Gefahr bei einem Programm ohne WENN ist, dass es losläuft wenn die CCU neu startet.
Stell dir mal vor es gibt einen kurzen Stromausfall, die CCU fährt wieder hoch und startet dabei das Programm ohne WENN Bedingung.

Bei manchen Programmen mag das egal sein, aber bestimmt nicht bei allen. Zum Beispiel bei einen Alarmanlagen Steuerung, die auch die Türen verriegelt bzw. entriegelt. 

Also ich wäre hier sehr vorsichtig bei Programmen ohne WENN.

Viele Grüße
Chris

**Comment by meistermolli** [comment]
@Peyto79 danke für den Tipp.
Aber wie bekommt ihr dann die Programme in HAP ans laufen?

die wichtigen Programme dann doch mit dem Virtuellen Taster einbinden?

**Comment by thkl** [comment]
Ähm Programme werden mit o.ProgramExecute() ausgeführt, egal was für Bedingungen sie haben ?

**Comment by meistermolli** [comment]
> Ähm Programme werden mit o.ProgramExecute() ausgeführt, egal was für Bedingungen sie haben ?

Ich habe in dem anderen issue das Programm so geprüft wie du dort beschrieben hattest.
Dort waren aber die Bedingungen gesetzt. Taster des WGC oder Virtueller Taster. (Bezog sich auf die Versuche 1. / 2.

3. habe ich mit dem Script Befehl nicht geprüft, da diese in homekit starten.


@thkl Kanndt du hier mal einen Screenshot von einem Programm einstellen, das bei dir über HomeKit funktioniert?
